ros2bag/ros2bag/verb/play.py

```diff
@@ -196,11 +198,41 @@ def main(self, *, args):  # noqa: D102
             topic_remapping.append('--remap')
             topic_remapping.append(remap_rule)
 
-        storage_options = StorageOptions(
-            uri=args.bag_path,
-            storage_id=args.storage,
-            storage_config_uri=storage_config_file,
-        )
+        # Do not allow using both positional arg and --input option for input bags
+        if args.bag_path and args.input:
+            return print_error(
+                'do not mix the [bag_path] positional argument and the -i,--input option; '
+                'for multiple input bags, use -i,--input multiple times')
+        # Add bag from optional positional arg, then bag(s) from optional flag
+        storage_options = []
+        if args.bag_path:
+            storage_options.append(StorageOptions(
+                uri=args.bag_path,
+                storage_id=args.storage,
+                storage_config_uri=storage_config_file,
+            ))
+            if args.storage:
+                print(print_warn('--storage option is deprecated, use -i,--input to '
+                                 'provide an input bag with a specific storage ID'))
```

Comment:
same as https://github.com/ros2/rosbag2/pull/1873#discussion_r1866608189

Reply:
- Well, I was thinking about it from the very beginning. And my opinion I would let it go as is. 
Because this is just a warning we are not removing anything. And it will stimulate others to use a new API as we have in the Rolling. It will be more uniform and more easily to handle and support in the future.

Reply:
yeah that sounds good, I just wanted to point it out just in case.